src/bijectors/corr.jl

```diff
@@ -0,0 +1,118 @@
+# See stan doc for parametrization method:
+# https://mc-stan.org/docs/2_23/reference-manual/correlation-matrix-transform-section.html
+# (7/30/2020) their "manageable expression" is wrong...
+
+struct CorrBijector <: Bijector{2} end
+
+function (b::CorrBijector)(x::AbstractMatrix{<:Real})    
+    w = cholesky(x).U + zero(x) # convert to dense matrix
+    r = _link_chol_lkj(w) 
+    return r
+end
+
+(b::CorrBijector)(X::AbstractArray{<:AbstractMatrix{<:Real}}) = map(b, X)
+
+function (ib::Inverse{<:CorrBijector})(y::AbstractMatrix{<:Real})
+    w = _inv_link_chol_lkj(y)
+    return w' * w
+end
+(ib::Inverse{<:CorrBijector})(Y::AbstractArray{<:AbstractMatrix{<:Real}}) = map(ib, Y)
+
+
+function logabsdetjac(::Inverse{CorrBijector}, y::AbstractMatrix{<:Real})
+    @assert size(y, 1) == size(y, 2)
+    K = size(y, 1)
+    
+    left = zero(eltype(y)) # Initial summand may make looping looks weird :(
+    @inbounds for j=2:K, i=1:(j-1)
+        # left += (K-i-1) * log(1 - tanh(y[i, j])^2) # lacks numerically stable
+        left += (K-i-1) * 2 * (log(2) - log(exp(y[i,j]) + exp(-y[i,j])))
+    end
+    
+    right = zero(eltype(y))
+    @inbounds for j=2:K, i=1:(j-1)
+        right += log(cosh(y[i, j])^2)
+    end
+    
+    return left / 2 - right
+end
+function logabsdetjac(b::CorrBijector, X::AbstractMatrix{<:Real})
+    return -logabsdetjac(inv(b),(b(X))) # It may be more efficient if we can use un-contraint value to prevent call of b
```

Comment:
The comment is not clear to me. In any case, we should try to stick to the limit of 92 characters per line and put it above the code.
```suggestion
    return -logabsdetjac(inv(b), b(X))
```

Reply:
Stan's doc lists only the inverse `logabsdetjac`. I think that they just calculate `logabsdet_inv(b_X)` instead of `logabsdet_inv(b(X))` (an extra `b` call. It's even more inefficient considering that `b` call cholesky decomposition), since they keep the un-constraint value `b_X` to that phase.

On the other hand, I found Turing's usage of `logpdf_with_trans` makes it impossible to remove the extra cost.

Reply:
Added explanation.